modules/bmc/ipmi.rb

```diff
@@ -18,6 +18,7 @@ def self.logger=(log)
 
       # returns boolean true if the specified provider is installed
       def self.installed?(provider)
+        return false if provider.empty?
```

Comment:
This does not belong here, I would not mind usually but this looks like a big refactoring/change and it can be delayed or whatever.

Reply:
This was needed to make the tests pass. Once you remove the Puppet gem, BMC tests start to fail. https://github.com/theforeman/smart-proxy/pull/637#issuecomment-499573545

I don't mind submitting it as a separate PR so we can review/merge it first.